Core/Src/can_handler.c

```diff
@@ -1,21 +1,27 @@
 #include "analyzer.h"
 #include "ringbuffer.h"
 #include "can_handler.h"
+#include "stateMachine.h"
 
 ringbuffer_t* can1_rx_queue = NULL;
 ringbuffer_t* can2_rx_queue = NULL;
+nc_fault_collection_t* nc_fault_collection = NULL;
```

Comment:
nit picking but, no need to make this a pointer. Right now, no memory is allocated for it so it would fail, but rather than adding a malloc() for it just make this a direct struct object